### PR TITLE
Add color overlay to new dashboard

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12391
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/Database/Models/UserSettingsTests.swift
* Core/CoreTests/Use Cases/GetUserSettingsTests.swift
* TestsFoundation/TestsFoundation/Fixtures/Model/UserSettingsFixture.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-12391](https://instructure.atlassian.net/browse/MBL-12391)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.62% | 93.6% | 0.03%
**Student** | 92.3% | 92.3% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.71% | 92.71% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Dashboard/Views/DashboardCourseCell.swift [hunk 1]
```diff
@@ -84,16 +85,45 @@ class DashboardCourseCell: UICollectionViewCell {
...
+
+        guard let optionsButton = optionsButton else {
+            return
+        }
+        let circleLayer = CALayer()
+        circleLayer.frame = CGRect(x: 4, y: 4, width: optionsButton.bounds.width - 8, height: optionsButton.bounds.height - 8)
+        circleLayer.backgroundColor = color.cgColor
+        circleLayer.cornerRadius = circleLayer.frame.width / 2
+        optionsButton.layer.insertSublayer(circleLayer, below: optionsButton.imageView?.layer)
+        optionsButtonCircle = circleLayer
```
Comment (vanwagonet) [inline Student/Student/Dashboard/Views/DashboardCourseCell.swift]: do we need to create a new layer, or can the button just have a background color & it's normal layer get rounded?

Comment (shichongrui) [reply Student/Student/Dashboard/Views/DashboardCourseCell.swift]: I initially went that route but the problem was that the button's size was larger than what we wanted the visual size of the circle to be.

Comment (vanwagonet) [reply Student/Student/Dashboard/Views/DashboardCourseCell.swift]: it likely doesn't need to be any bigger than the circle is. We probably were just going for a nice hit area, which may not be necessary.

Comment (shichongrui) [reply Student/Student/Dashboard/Views/DashboardCourseCell.swift]: I'll chat with David to see what he thinks. The circle is only 26x26 which is a bit small, and the button is 34x34 which still seems small but I'll see what he thinks.

Comment (shichongrui) [reply Student/Student/Dashboard/Views/DashboardCourseCell.swift]: My intent was to maintain the larger hit target while visually having it match the designs.

**Pull request merged**